core/src/main/java/io/questdb/cutlass/parquet/SerialParquetExporter.java

```diff
@@ -0,0 +1,409 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cutlass.parquet;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.PartitionBy;
+import io.questdb.cairo.TableReader;
+import io.questdb.cairo.TableToken;
+import io.questdb.cairo.sql.ExecutionCircuitBreaker;
+import io.questdb.cairo.sql.PartitionFormat;
+import io.questdb.cutlass.text.CopyExportResult;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.table.parquet.ParquetCompression;
+import io.questdb.griffin.engine.table.parquet.PartitionDescriptor;
+import io.questdb.griffin.engine.table.parquet.PartitionEncoder;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.Chars;
+import io.questdb.std.Files;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.Misc;
+import io.questdb.std.Numbers;
+import io.questdb.std.datetime.DateLocaleFactory;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+import io.questdb.std.str.Utf8StringSink;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+
+import static io.questdb.cairo.CairoException.TABLE_DOES_NOT_EXIST;
+import static io.questdb.cairo.TableUtils.createDirsOrFail;
+
+public class SerialParquetExporter implements Closeable {
+    private static final Log LOG = LogFactory.getLog(SerialParquetExporter.class);
+    private final CairoEngine cairoEngine;
+    private final CairoConfiguration configuration;
+    private final CharSequence copyExportRoot;
+    private final StringSink exportPath = new StringSink(128);
+    private final FilesFacade ff;
+    private final Path fromParquet;
+    private final Utf8StringSink nameSink = new Utf8StringSink();
+    private final Path tempPath;
+    private final Path toParquet;
+    private ExecutionCircuitBreaker circuitBreaker;
+    private int numOfFiles;
+    private PhaseStatusReporter statusReporter;
+    private CopyExportRequestTask task;
+
+    public SerialParquetExporter(CairoEngine engine) {
+        this.cairoEngine = engine;
+        this.configuration = engine.getConfiguration();
+        this.ff = this.configuration.getFilesFacade();
+        this.copyExportRoot = this.configuration.getSqlCopyExportRoot();
```

Comment:
what is the benefit of caching this value and requiring server restart when it is changed?